idevice/src/services/misagent.rs

```diff
@@ -2,200 +2,316 @@
 //!
 //! Provides functionality for interacting with the misagent service on iOS devices,
 //! which manages provisioning profiles and certificates.
+//!
+//! Based on libimobiledevice implementation from SideStore
 
-use log::warn;
-use plist::Dictionary;
+use log::{debug, warn};
+use plist::{Dictionary, Value};
 
-use crate::{obf, Idevice, IdeviceError, IdeviceService, RsdService};
+use crate::{lockdown::LockdownClient, obf, Idevice, IdeviceError, IdeviceService};
 
-/// Client for interacting with the iOS misagent service
-///
-/// The misagent service handles:
-/// - Installation of provisioning profiles
-/// - Removal of provisioning profiles
-/// - Querying installed profiles
-pub struct MisagentClient {
-    /// The underlying device connection with established misagent service
-    pub idevice: Idevice,
+/// Error codes returned by misagent service
```

Comment:
I do not agree with creating another error enum. Why not just extend the Idevice one?